### ISSUE TITLE
[JIT][script] PythonOp may introduce requires_grad

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#7438]
GraphExecutor cannot assume there is no gradient if there's a PythonOp in the graph

Example model:

```
class VAE(torch.jit.ScriptModule):
    def __init__(self):
        super(VAE, self).__init__()

        # self.fc1 = nn.Linear(784, 400)
        self.fc1 = nn.Linear(784, 20)
        self.fc21 = nn.Linear(400, 20)
        self.fc22 = nn.Linear(400, 20)
        self.fc3 = nn.Linear(20, 400)
        self.fc4 = nn.Linear(400, 784)

    __constants__ = ['training']

    @torch.jit.script_method
    def encode(self, x):
        # h1 = F.relu(self.fc1(x))
        # return self.fc21(h1), self.fc22(h1)
        h1 = self.fc1(x)
        return h1, h1

    @torch.jit.script_method
    def reparameterize(self, mu, logvar):
        retval = 0
        if self.training:
            std = torch.exp(0.5*logvar)
            eps = torch.randn_like(std)
            retval = eps.mul(std) + mu
        else:
            retval = mu
        return retval

    @torch.jit.script_method
    def decode(self, z):
        h3 = F.relu(self.fc3(z))
        return F.sigmoid(self.fc4(h3))

    # @torch.jit.script_method
    def forward(self, x):
        print('x.requires_grad', x.requires_grad)
        mu, logvar = self.encode(x.view(-1, 784))
        print(self.encode.graph)
        print('mu.requires_grad', mu.requires_grad)
        z = self.reparameterize(mu, logvar)
        print('z.requires_grad', z.requires_grad)
        retvals = (self.decode(z), mu, logvar)
        print('retvals[0].requires_grad', retvals[0].requires_grad)
        return retvals
```

In `encode`, `fc1` introduces `requires_grad`, but running in the interpreter does not have this behavior because `fc1` is a PythonOp calling out to `nn.Linear`.

Shouldn't use fact that we're working on Variables as a signal that we need grad. Remove `values_are_variables`.
